src/ast/rewriter/seq_rewriter.cpp

```diff
@@ -1912,11 +1967,63 @@ br_status seq_rewriter::mk_seq_replace_all(expr* a, expr* b, expr* c, expr_ref&
         result = str().mk_concat(strs, a->get_sort());
         return BR_REWRITE_FULL;
     }
-    // TBD: add case when a, b are concatenation of units that are values.
-    // in this case we can use a similar loop as for strings.
+    expr_ref_vector a_vals(m());
+    expr_ref_vector b_vals(m());
+    if (try_get_unit_values(a, a_vals) && try_get_unit_values(b, b_vals))
+    {
```

Comment:
{ after if is not on separate line (contrast to typical C# conventions).

Reply:
fixed this and the ones above in the following commit